src/facade/dragonfly_connection.h

```diff
@@ -54,14 +54,21 @@ class Connection : public util::Connection {
 
   struct PubMessage {
     // if empty - means its a regular message, otherwise it's pmessage.
-    std::string_view pattern;
-    std::string_view channel;
-    std::shared_ptr<const std::string> message;  // ensure that this message would out live passing
-                                                 // between different threads/fibers
+    std::string pattern;
+    std::shared_ptr<std::string> channel;
+    std::shared_ptr<std::string> message;  // ensure that this message would out live passing
```

Comment:
In the future we should probably just wrap this whole type into one shared ptr

Reply:
yeah, it's not possible due to `pattern` field that is unique for each psubscriber.
we could only wrap (message,channel) together